Store Creation: Track when creating new site and enabling free trial completes

itsmeichigo opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #10030 
<!-- Id number of the GitHub issue this PR addresses. -->

## Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
This PR follows the internal discussion [p1687174598489079-slack-C03L1NF1EA3] to add a new tracking event to determine stores created from mobile apps.

## Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->
- Log in to a WPCom site.
- Switch to Menu tab and open the store picker.
- Select Add a store -> Create a new store.
- Complete the store creation flow. After tapping the Start free trial button, notice in Xcode console: `🔵 Tracked site_creation_free_trial_created_success, properties: [AnyHashable("initial_domain"): "...", AnyHashable("blog_id"):<blog_id>, AnyHashable("is_wpcom_store"): true]`

## Screenshots
<!-- Include before and after images or gifs when appropriate. -->
N/A

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

itsmeichigo committed:
Add new event name

itsmeichigo committed:
Update wooanalyticsevent+storecreation to include new event

itsmeichigo committed:
Track store creation success

selanthiraiyan reviewed (approved):
:shipit: 

#### Non blocking observations

--- 
When creating a new site during login, from the site picker before selecting site.

```Tracked site_creation_free_trial_created_success, properties: [AnyHashable("initial_domain"): "changed-public-link-puppy.wordpress.com/"]```
* Please note the slash at the end of the `initial_domain` URL.

---

When creating a new site during login, from Menu -> site picker

```Tracked site_creation_free_trial_created_success, properties: [AnyHashable("initial_domain"): "changed-public-link-lover.wordpress.com/", AnyHashable("is_wpcom_store"): true, AnyHashable("blog_id"): 123456789]```
* Please note the slash at the end of the `initial_domain` URL.
* Just FYI, please note that the `blog_id` belongs to the already logged in store and not the newly created store. 
---

itsmeichigo committed:
Use site slug to track domain name

itsmeichigo committed:
Update custom property blog_id to new_site_id

itsmeichigo commented:
Great catch @selanthiraiyan! The issue with `blog_id` is that it was overwritten by the default property, so I changed the property name to `new_site_id`. I also switched to using site slug instead of site URL to avoid the trailing slash. Please take another look!

selanthiraiyan reviewed (approved):
Thanks for the updates, Huong! This works as expected now. 🎉

itsmeichigo commented:
@spencertransier Could you help with a new beta build please? Thanks!

spencertransier commented:
This PR is included in the `14.1.0.2` beta build.
